benchmark/cmd/seidb/main.go

```diff
@@ -0,0 +1,263 @@
+	generateCmd.Flags().IntVar(&version, "version", 0, "db version")
+
+	benchmarkWriteCmd.Flags().StringVar(&dbBackend, "db-backend", "", "DB Backend")
+	benchmarkWriteCmd.Flags().StringVar(&rawKVInputDir, "raw-kv-input-dir", "", "Input Directory for benchmark which contains the raw kv data")
+	benchmarkWriteCmd.Flags().StringVar(&outputDir, "output-dir", "", "Output Directory")
+	benchmarkWriteCmd.Flags().IntVar(&concurrency, "concurrency", 1, "Concurrency while writing to db")
+	benchmarkWriteCmd.Flags().IntVar(&batchSize, "batch-size", 1, "batch size for db writes")
+	benchmarkWriteCmd.Flags().IntVar(&numVersions, "num-versions", 1, "number of versions in db")
+
+	benchmarkReadCmd.Flags().StringVar(&dbBackend, "db-backend", "", "DB Backend")
+	benchmarkReadCmd.Flags().StringVar(&rawKVInputDir, "raw-kv-input-dir", "", "Input Directory for benchmark which contains the raw kv data")
+	benchmarkReadCmd.Flags().StringVar(&outputDir, "output-dir", "", "Output Directory which contains db")
+	benchmarkReadCmd.Flags().IntVar(&concurrency, "concurrency", 1, "Concurrency while reading from db")
+	benchmarkReadCmd.Flags().Int64Var(&maxOps, "max-operations", 1000, "Max operations to run")
+	benchmarkReadCmd.Flags().IntVar(&numVersions, "num-versions", 1, "number of versions in db")
+
+	benchmarkForwardIterationCmd.Flags().StringVar(&dbBackend, "db-backend", "", "DB Backend")
+	benchmarkForwardIterationCmd.Flags().StringVar(&rawKVInputDir, "raw-kv-input-dir", "", "Input Directory for benchmark which contains the raw kv data")
+	benchmarkForwardIterationCmd.Flags().StringVar(&outputDir, "output-dir", "", "Output Directory which contains db")
+	benchmarkForwardIterationCmd.Flags().IntVar(&concurrency, "concurrency", 1, "Concurrency while reading from db")
+	benchmarkForwardIterationCmd.Flags().Int64Var(&maxOps, "max-operations", 1000, "Max operations to run")
+	benchmarkForwardIterationCmd.Flags().IntVar(&iterationSteps, "iteration-steps", 10, "Number of steps to run per iteration")
+	benchmarkForwardIterationCmd.Flags().IntVar(&numVersions, "num-versions", 1, "number of versions in db")
+
+	benchmarkReverseIterationCmd.Flags().StringVar(&dbBackend, "db-backend", "", "DB Backend")
+	benchmarkReverseIterationCmd.Flags().StringVar(&rawKVInputDir, "raw-kv-input-dir", "", "Input Directory for benchmark which contains the raw kv data")
+	benchmarkReverseIterationCmd.Flags().StringVar(&outputDir, "output-dir", "", "Output Directory which contains db")
+	benchmarkReverseIterationCmd.Flags().IntVar(&concurrency, "concurrency", 1, "Concurrency while reading from db")
+	benchmarkReverseIterationCmd.Flags().IntVar(&iterationSteps, "iteration-steps", 10, "Number of steps to run per iteration")
+	benchmarkReverseIterationCmd.Flags().IntVar(&numVersions, "num-versions", 1, "number of versions in db")
+}
+
+func main() {
+	if err := rootCmd.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+func generate(cmd *cobra.Command, args []string) {
```

Comment:
Would be good to put each subcommand into its own .go file

Reply:
Updated